src/main/scala/utility/SubVec.scala

```diff
@@ -0,0 +1,58 @@
+/***************************************************************************************
+* Copyright (c) 2024 Beijing Institute of Open Source Chip (BOSC)
+* Copyright (c) 2024 Institute of Computing Technology, Chinese Academy of Sciences
+*
+* XiangShan is licensed under Mulan PSL v2.
+* You can use this software according to the terms and conditions of the Mulan PSL v2.
+* You may obtain a copy of Mulan PSL v2 at:
+*          http://license.coscl.org.cn/MulanPSL2
+*
+* THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+* EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+* MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+*
+* See the Mulan PSL v2 for more details.
+***************************************************************************************/
+
+package utility
+
+import chisel3._
+
+object SubVec{
+  // Rem: SubSeq = (Seq % Modulus)(Remainder)
+  // NOTE: must be divisible
+  def getRem[T <: Data](in: Seq[T], mod: Int, rem: Int): Seq[T] = {
+    require(in.size % mod == 0)
+    require(mod > rem)
+    (0 until in.size / mod).map(i => {in(mod * i + rem)})
+  }
+  def getRems[T <: Data](in: Seq[T], mod: Int): Seq[Seq[T]] = {
+    (0 until mod).map { rem => getRem(in, mod, rem) }
+  }
+
+  // RemWithExpand: SubSeqWithExpand = (Seq % Modulus expanded)(Remainder)
+  def getRemWithExpand(in: Seq[Bool], mod: Int, rem: Int): Seq[Bool] = {
+    require(mod > rem)
+    val subSize = in.size / mod + 1
+    (0 until subSize).map {i =>
+      val idx = mod * i + rem
+      if (idx < in.size) in(idx) else false.B
+    }
+  }
+  def getRemWithExpands(in: Seq[Bool], mod: Int): Seq[Seq[Bool]] = {
```

Comment:
Can you change the name of the function？For example, getRem can be called do_unzip or do_extract, and getMaskRem can be called do_zip or do_insert.

Reply:
The names 'zip' or 'extract' feel unintuitive, so the original names are retained here.